presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnectionConfig.java

```diff
@@ -39,26 +49,33 @@
 
     private final HostAndPort address;
     private final URI uri;
+    private final Properties connectionProperties;
 
     private String catalog;
     private String schema;
 
     private final boolean useSecureConnection;
 
-    public PrestoDriverUri(String url)
+    private static final Map<ConnectionProperty, BiConsumer<HttpClientConfig, String>> CLIENT_SETTERS = ImmutableMap.of(
```

Comment:
It also makes it one touch to add other URL parameters/Properties to the HttpClientConfig (like the keystore if you need it; were you able to use the truststore parameters or do you need me to add keystore support too?).

Making HttpClientCreator a POJO and having a factory method introduces more touch points:
1. the POJO
2. the equals method
3. the hashCode method
4. the factory method